moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame.cpp

```diff
@@ -246,10 +248,33 @@ void MotionPlanningFrame::allowExternalProgramCommunication(bool enable)
   }
 }
 
+void MotionPlanningFrame::fillPlanningGroupOptions()
+{
+  ui_->planning_group_combo_box->clear();
+  const robot_model::RobotModelConstPtr& kmodel = planning_display_->getRobotModel();
+  std::string group = planning_display_->getCurrentPlanningGroup();
+
+  const std::vector<std::string> groups = kmodel->getJointModelGroupNames();
+  size_t group_index = 0;
+  size_t current_index = 0;
+  for (const std::string group_name : groups)
+  {
+    ui_->planning_group_combo_box->addItem(QString::fromStdString(group_name));
+
+    if (group_name == group)
+      group_index = current_index;
+    ++current_index;
+  }
+
+  ui_->planning_group_combo_box->setCurrentIndex(group_index);  // currently chosen group
+}
+
 void MotionPlanningFrame::fillStateSelectionOptions()
 {
-  ui_->start_state_selection->clear();
-  ui_->goal_state_selection->clear();
+  ui_->start_state_combo_box->blockSignals(true);
```

Comment:
There is a nice utility class taking care of unblocking signals automatically: [QSignalBlocker](http://doc.qt.io/qt-5/qsignalblocker.html).